faust/sensors/prometheus.py

```diff
@@ -0,0 +1,317 @@
+                                assignor: PartitionAssignorT,
+                                state: typing. Dict) -> None:
+        """Partition assignor completed assignment."""
+        super().on_assignment_completed(assignor, state)
+        self.assignment_operations.labels(operation=self.COMPLETED).inc()
+        self.assign_latency.observe(
+            self.ms_since(state['time_start']))
+
+    def on_rebalance_start(self, app: AppT) -> typing.Dict:
+        """Cluster rebalance in progress."""
+        state = super().on_rebalance_start(app)
+        self.total_rebalances.inc()
+
+        return state
+
+    def on_rebalance_return(self, app: AppT, state: typing.Dict) -> None:
+        """Consumer replied assignment is done to broker."""
+        super().on_rebalance_return(app, state)
+        self.total_rebalances.dec()
+        self.total_rebalances_recovering.inc()
+        self.revalance_done_consumer_latency.observe(
+            self.ms_since(state['time_return']))
+
+    def on_rebalance_end(self, app: AppT, state: typing.Dict) -> None:
+        """Cluster rebalance fully completed (including recovery)."""
+        super().on_rebalance_end(app, state)
+        self.total_rebalances_recovering.dec()
+        self.revalance_done_latency.observe(
+            self.ms_since(state['time_end']))
+
+    def count(self, metric_name: str, count: int = 1) -> None:
+        """Count metric by name."""
+        super().count(metric_name, count=count)
+        self.count_metrics_by_name.labels(metric=metric_name).inc(count)
+
+    def on_tp_commit(self, tp_offsets: TPOffsetMapping) -> None:
+        """Call when offset in topic partition is committed."""
+        super().on_tp_commit(tp_offsets)
+        for tp, offset in tp_offsets.items():
+            self.topic_partition_offset_commited.labels(
+                topic=tp.topic, partition=tp.partition).set(offset)
+
+    def track_tp_end_offset(self, tp: TP, offset: int) -> None:
+        """Track new topic partition end offset for monitoring lags."""
+        super().track_tp_end_offset(tp, offset)
+        self.topic_partition_end_offset.labels(
+            topic=tp.topic, partition=tp.partition).set(offset)
+
+    def on_web_request_end(self,
+                           app: AppT,
+                           request: web.Request,
+                           response: typing.Optional[web.Response],
+                           state: typing.Dict,
+                           *,
+                           view: web.View = None) -> None:
+        """Web server finished working on request."""
+        super().on_web_request_end(app, request, response, state, view=view)
+        status_code = int(state['status_code'])
+        self.http_status_codes.labels(status_code=status_code).inc()
+        self.http_latency.observe(
+            self.ms_since(state['time_end']))
+
+    def start_client(self) -> None:
+        """Start promethues client."""
+        return start_wsgi_server(self.port)
```

Comment:
Thanks for the feedback, really appreciated. 

Regarding to `start_wsgi_server`, I was a bit worried as well because even though the `wsgi` server will run in a different thread, it is blocking, and that is not good at all. The idea behind was to server the metrics using a different `port` to don't mix `web views` with `metrics`, but I think was not a good idea at all. I will update the code to use the current `aiohttp` application to serve the metrics with a configurable `url/pattern`, replacing the `start_client` method with `expose_metrics` method which will create the metrics handler and attach it to the `web router`.

Regarding to `Summary` suggestion, I think is a good idea. Thanks.

Reply:
I believe you have to set this parameter to have aiohttp run in a separate thread...
```
web_in_thread
New in version 1.5.

type
bool

default
False

Run the web server in a separate thread.

Use this if you have a large value for stream_buffer_maxsize and want the web server to be responsive when the worker is otherwise busy processing streams.

Note
Running the web server in a separate thread means web views and agents will not share the same event loop.
```